### PR TITLE
[Maistra-1244] [Maistra-1245] Make buildinfo update version

### PR DESCRIPTION
**Pull request by brian-avery**
Thought we already had this, but update.sh isn't updating buildinfo. This PR fixes it.

### CODE REVIEW

**Comment by brian-avery** [comment]
@knrc Currently, update.sh doesn't update the version in buildinfo. This caused the version not to be updated in the 1.1 product update. Is it ok to merge for 1.1 or would you like to wait until post 1.1? We can always update that manually for now.

**Comment by brian-avery** [comment]
This functionality will be moved to the spec file.

**Comment by brian-avery** [comment]
Moved the changes to the spec file and confirmed that this doesn't make the version problem any worse. 

`version.BuildInfo{Version:"unknown", GitRevision:"unknown", GolangVersion:"go1.14", BuildStatus:"unknown", GitTag:"unknown"}`

**Review by knrc — APPROVED** [review]

**Review by knrc — CHANGES_REQUESTED** [review]
The names are incorrect, I'll add more details into the bug report

**Comment by brian-avery** [comment]
Updated the names of buildinfo parameters in here as well, which fixed maistra-1244 in the process. 

version.BuildInfo{Version:"1.1.0", GitRevision:"b0f47f26fa67a352fc95db7ecd2af902d7879221", GolangVersion:"go1.14", BuildStatus:"Clean", GitTag:"unknown"}

**Review by knrc — APPROVED** [review]

File: istio/istio.spec [hunk 1]
```diff
@@ -376,6 +376,8 @@ mkdir -p ISTIO/src/istio.io/istio
 tar zxf %{SOURCE0} -C ISTIO/src/istio.io/istio --strip=1
 
 cp %{SOURCE1} ISTIO/src/istio.io/istio/.istiorc.mk
+
+sed -i "s/istio.io\/pkg\/version.buildVersion=.*/istio.io\/pkg\/version.buildVersion="%{version}"/" %{SOURCE2}
```
Comment (jwendell) [inline istio/istio.spec]: This is not necessary. The changes in `buildinfo` file are enough, and changing the version number it is part (or should be) of the building process.

Comment (brian-avery) [reply istio/istio.spec]: @jwendell This updates buildinfo on the build so the user isn't expected to manually update it. We missed that in 1.1, so this fixes it.

Comment (brian-avery) [reply istio/istio.spec]: This is the fix for maistra-1245, not 1244.

Comment (knrc) [reply istio/istio.spec]: +1 the definitive version/release is defined within the spec file so it should be updating buildinfo as part of the build process.

**Comment by brian-avery** [comment]
I added an additional commit to this to remove the docker hub. This field is no longer printed in the 1.4 version info.

**Review by rcernich — APPROVED** [review]

**Pull request merged**